services/community-site/src/services/useApi.ts

```diff
@@ -90,15 +90,22 @@ const useApi = (baseUrl = process.env.REACT_APP_API_HOST) => {
   }, []);
 
   const post = useCallback(
-    async <T>(url: string, data: Record<string, unknown> | FormData, includeToken = false) => {
+    async <T>(
+      url: string,
+      data: Record<string, unknown> | FormData,
+      includeToken = false,
+      shouldLoad = true,
+    ) => {
       const responseHandler = new TypedResponseHandler<T>();
       const options = getOptions(includeToken);
-      startLoading!();
+      if (shouldLoad) startLoading!();
```

Comment:
I think we need to remove this from here.
We should have a component state for definining when the inside of the stats are loading or not based on which we should show the Skeleton or not. The APIs should trigger as before the loading as they only show a spinner.